From a5e27d0d6a19f5dc2eb6fd2b6757378e0a7df6d9 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:14 -0400
Subject: [scsi] qla2xxx: Add handling for boot indication progress AENs for ISPFX00

Message-id: <1408391817-6389-27-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87950
O-Subject: [PATCH RHEL 7.1 e-stor 26/69] qla2xxx: Add handling for boot indication progress AENs for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Armen Baloyan <armen.baloyan@qlogic.com>

scsi commit id 0f8cdff567357fe7280963bc90e17e97dad995cd

Signed-off-by: Armen Baloyan <armen.baloyan@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_def.h b/drivers/scsi/qla2xxx/qla_def.h
index f2ac19e..f9c79a1 100644
--- a/drivers/scsi/qla2xxx/qla_def.h
+++ b/drivers/scsi/qla2xxx/qla_def.h
@@ -808,7 +808,7 @@ struct mbx_cmd_32 {
         Notification */
 #define MBA_FW_POLL_STATE 0x8600  /* Firmware in poll diagnostic state */
 #define MBA_FW_RESET_FCT 0x8502 /* Firmware reset factory defaults */
-
+#define MBA_FW_INIT_INPROGRESS 0x8500 /* Firmware boot in progress */
 /* 83XX FCoE specific */
 #define MBA_IDC_AEN  0x8200  /* FCoE: NIC Core state change AEN */
 
diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index fec48bd..89cfe56 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -997,6 +997,9 @@ qlafx00_init_fw_ready(scsi_qla_host_t *vha)
    break;
 
   default:
+   if ((aenmbx & 0xFF00) == MBA_FW_INIT_INPROGRESS)
+    break;
+
    /* If fw is apparently not ready. In order to continue,
     * we might need to issue Mbox cmd, but the problem is
     * that the DoorBell vector values that come with the
-- 
1.7.1